tests/integ/modin/binary/test_binary_op.py

```diff
@@ -2517,3 +2517,23 @@ def test_binary_bitwise_op_on_df(df1, df2, func, expected):
 
     snow_ans = func(snow_df1, snow_df2)
     assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(snow_ans, expected)
+
+
+@sql_count_checker(query_count=2)
+@pytest.mark.parametrize(
+    "func",
+    [
+        lambda df: df + df[0],
+        lambda df: df - df[0],
+        lambda df: df[0] + df,
+        lambda df: df[0] - df,
```

Comment:
what happend if the length does not match between the series and df?

Reply:
We behave the same way as pandas in those cases, there are already tested by `test_binary_op_between_df_and_list_like_of_different_lengths_axis_1` and `test_binary_op_between_series_of_different_lengths`

Reply:
should throw an error, I think we capture this in other tests.

Reply:
Not necessarily an error (in some cases yes, but for example this one computes in pandas):

```
In [2]: pandas.Series([5, 6]) + pandas.DataFrame([1, 7, 8])
Out[2]: 
    0   1
0   6 NaN
1  12 NaN
2  13 NaN
```

Reply:
This is not exactly identical to NumPy broadcasting, right? At ponder, we decided to deviate from pandas here and implement NumPy broadcasting because we felt it's more useful.